spring-cloud-starter-stream-processor-tensorflow/src/main/java/org/springframework/cloud/stream/app/tensorflow/processor/TensorflowProcessorProperties.java

```diff
@@ -46,6 +46,22 @@
 	 */
 	private int outputIndex = 0;
 
+	/**
+	 * Controls if the inference result is stored in the {@link org.springframework.messaging.Message} payload
+	 * or header.
+	 * By default (false) the result is stored in the payload. When set to true, the result will be
+	 * stored in the header using the resultHeaderName value.
+	 */
+	private boolean saveResultInHeader = false;
+
+	/**
+	 * Applicable only for `saveResultInHeader=true`, this property holds the header name used to store
+	 * the inference result. The default name is `tf_result`.
+	 * If multiple tensorflow processors are connected in a single pipeline then make sure they have
+	 * distinct resultHeaderName values to avoid overriding.
+	 */
+	private String resultHeaderName = "tf_result";
```

Comment:
@artembilan,  actually i've been thinking along same lines :) But i am  not convinced yet that we should implicitly (e.g. by default) send the result in the header.  
Furthermore considering the Binder peculiarities, such as LongString on Rabbit or the fact that you have to explicitly whitelist the custom headers on Kafka, makes me feel less confident in the header-first approach ;)
Maybe we need more time to get better understanding of the proper usage patterns. We need to try more TF models and use cases :)